diff -Naur a/data/com.github.johnfactotum.Foliate.desktop.in b/data/com.github.johnfactotum.Foliate.desktop.in
--- a/data/com.github.johnfactotum.Foliate.desktop.in	2023-06-08 11:49:23.723100833 +0530
+++ b/data/com.github.johnfactotum.Foliate.desktop.in	2023-06-12 21:52:37.437293364 +0530
@@ -5,7 +5,7 @@
 Categories=Office;Viewer;
 MimeType=application/epub+zip;application/x-mobipocket-ebook;application/vnd.amazon.mobi8-ebook;application/x-fictionbook+xml;application/x-zip-compressed-fb2;application/vnd.comicbook+zip;
 Exec=foliate %U
-Icon=com.github.johnfactotum.Foliate
+Icon=snap.foliate.com.github.johnfactotum.Foliate
 Terminal=false
 Type=Application
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
diff -Naur a/data/com.github.johnfactotum.Foliate.metainfo.xml.in b/data/com.github.johnfactotum.Foliate.metainfo.xml.in
--- a/data/com.github.johnfactotum.Foliate.metainfo.xml.in	2023-06-08 11:49:23.723100833 +0530
+++ b/data/com.github.johnfactotum.Foliate.metainfo.xml.in	2023-06-12 21:52:27.361334434 +0530
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
     <id>com.github.johnfactotum.Foliate</id>
+    <icon type="stock">snap.foliate.com.github.johnfactotum.Foliate</icon>
     <metadata_license>CC0-1.0</metadata_license>
     <project_license>GPL-3.0-or-later</project_license>
     <name>Foliate</name>
